up-l1/zenoh.adoc

```diff
@@ -20,99 +20,249 @@ SPDX-License-Identifier: Apache-2.0
+a| *MUST* be set to the bytes representing the protobuf encoding of the UAttributes object as defined by link:../up-core-api/uprotocol/v1/uattributes.proto
+
 |===
 
-Note that the only supported version now is 0x01.
 
 ==== Message Type
 
-There are 4 kinds of message types in uProtocol (publish, notification, request, response), all of said messages are sent using the zenoh `put()` API meaning we shall use the pub/sub infrastructure only of zenoh and not the queryable APIs.
 
+[.specitem,oft-sid="dsn~up-transport-zenoh-message-type-mapping~1",oft-needs="impl,utest"]
+--
+All types of uProtocol messages *MUST* be transferred using Zenoh's pub/sub API only. Zenoh's Queryable API *MUST NOT* be used.
+--
 
-==== Priority Mapping (uProtocol to Zenoh):
+==== Message Priority
+
+[.specitem,oft-sid="dsn~up-transport-zenoh-message-priority-mapping~1",oft-needs="impl,utest"]
+--
+The uProtocol and Zenoh message priority levels *MUST* be mapped according to the following table:
 
 [cols="1,1"]
 |===
-| uProtocol priority | Zenoh Priority
+| uProtocol Priority | Zenoh Priority
+
+| `CS0` | `BACKGROUND`
+| `CS1` | `DATA_LOW`
+| `CS2` | `DATA`
+| `CS3` | `DATA_HIGH`
+| `CS4` | `INTERACTIVE_LOW`
+| `CS5` | `INTERACTIVE_HIGH`
+| `CS6` | `REAL_TIME`
+|===
+--
+
+=== Payload Mapping
+
+[.specitem,oft-sid="dsn~up-transport-zenoh-payload-mapping~1",oft-needs="impl,utest"]
+--
+A Zenoh message that is used to convey a uProtocol message *MUST* contain in its payload the unaltered value of the UMessage's _payload_ field.
+--
+
+== Zenoh Key Structure
+
+Message producers publish messages to the Zenoh network using _keys_. Other clients can indicate their interest in particular keys or patterns in order to receive the messages that are being published using matching keys.
+
+The Zenoh _key expression_ that is used to transfer a uProtocol message is derived from the message's `source` and `sink` attributes.
 
-| CS0 | BACKGROUND
-| CS1 | DATA_LOW
-| CS2 | DATA
-| CS3 | DATA_HIGH
-| CS4 | INTERACTIVE_LOW
-| CS5 | INTERACTIVE_HIGH
-| CS6 | REAL_TIME
+[.specitem,oft-sid="dsn~up-transport-zenoh-key-expr~1",oft-needs="impl,utest"]
+--
+The key expression of a Zenoh message containing a _Publish_ UMessage **MUST** consist of the following segments:
+
+`up/[source.authority]/[source.ue_type]/[source.ue_instance]/[source.ue_version]/[source.resource]/{}/{}/{}/{}/{}`
+
+The key expression of a Zenoh message containing a _Notification_, _RPC Request_ or _RPC Response_ UMessage **MUST** consist of the following segments:
+
+`up/[source.authority]/[source.ue_type]/[source.ue_instance]/[source.ue_version]/[source.resource]/[sink.authority]/[sink.ue_type]/[sink.ue_instance]/[sink.ue_version]/[sink.resource]`
+
+Please refer to <<UUri Encoding Rules>> for details regarding the encoding of the `source` and `sink` UUris into the topic's segments.
+--
+
+=== Examples
+
+The examples below assume that the local entity's authority name is `device1`.
```

Comment:
I'm okay with unifying the authority name, but we might need to align with the tests inside `up-transport-zenoh-rust` later.

Reply:
Yes, I can do that in the PR for amending the zenoh transport code with the specitem references ...

Reply:
Yes, I can do that in the PR for amending the zenoh transport code with the specitem references ...